src/features/broker-protection/comparison-functions.js

```diff
@@ -0,0 +1,449 @@
+            userParsedAddress.state === comparisonParsedAddress.state
+        ]
+
+        if (comparisons.every(Boolean)) {
+            return true
+        }
+
+        if (
+            !missingState &&
+      comparisonAddressFull.includes(userAddress.city) &&
+      comparisonAddressFull.includes(userAddress.state)
+        ) {
+            return true
+        }
+    }
+
+    return false
+}
+
+/**
+ * @param userAddresses
+ * @param foundAddresses
+ * @return {{cityFound, stateFound}|boolean}
+ */
+export function matchAddressFromAddressListCityState (userAddresses, foundAddresses) {
+    if (!userAddresses || userAddresses.length < 1 || !foundAddresses || foundAddresses.length < 1) {
+        return false
+    }
+
+    let cityFound, stateFound
+
+    for (const userAddress of userAddresses) {
+        const userCity = userAddress.city
+        const userState = userAddress.state
+
+        // for some reason when there is one line of addresses split by commas, it messes this up
+        // i.e. Chicago IL, Something Else IL, Asdf...
+        for (const possibleLocation of foundAddresses) {
+            cityFound = possibleLocation.city
+            stateFound = possibleLocation.state
+
+            if (isSameAddressCityState(userCity, userState, cityFound, stateFound)) {
+                return { cityFound, stateFound }
+            }
+        }
+    }
+
+    return false
+}
+
+/**
+ * @param addressList
+ * @param cityStateFound
+ * @return {boolean}
+ */
+export function matchAddressCityState (addressList, cityStateFound) {
+    if (!cityStateFound) { return false }
+
+    const cityFound = cityStateFound.split(',')[0]
+    const stateFound = cityStateFound.split(',')[1]
+
+    for (const userAddress of addressList) {
+        if (isSameAddressCityState(userAddress.city, userAddress.state, cityFound, stateFound)) {
+            return true
+        }
+    }
+
+    return false
+}
+
+/**
+ * @param city
+ * @param state
+ * @param comparisonCity
+ * @param comparisonState
+ * @return {boolean}
+ */
+export function isSameAddressCityState (city, state, comparisonCity, comparisonState) {
+    if (!city || !state || !comparisonCity || !comparisonState) { return false }
+
+    city = city.toLowerCase()?.trim()
+    comparisonCity = comparisonCity.toLowerCase()?.trim()
+    state = state.toLowerCase()?.trim()
+    comparisonState = comparisonState.toLowerCase()?.trim()
+
+    if ((city === comparisonCity) && (state === comparisonState)) { return true }
+
+    return false
+}
+
+export function capitalize (s) {
+    const words = s.split(' ')
+    const capitalizedWords = words.map(word => word.charAt(0).toUpperCase() + word.slice(1))
+    return capitalizedWords.join(' ')
+}
+
+export function getNicknames (name) {
+    if (name == null || name.trim() === '') { return [] }
+
+    name = name.toLowerCase()
+
+    // This comes from Removaly's list of common nicknames
+    const names = {
+        abby: ['abby', 'abigail'],
```

Comment:
Do we want this and the state abbreviations in this file? Seems like something we could kick to a constants directory so it's easier to find?

Reply:
agreed, nice idea